posthog/temporal/data_imports/pipelines/postgres/postgres.py

```diff
@@ -150,7 +152,14 @@ def _get_partition_bucket_size(cursor: psycopg.Cursor, schema: str, table_name:
     if result is None or len(result) == 0 or result[0] is None:
         return None
 
-    return int(result[0])
+    partition_size = int(result[0])
+    total_rows = int(result[1])
+    partition_count = math.floor(total_rows / partition_size)
```

Comment:
I _think_ you might have been able to simplify this to (pseudo code) `partition_count = table_size / bytes_per_partition` to make things a bit more efficient